src/EditorFeatures/Core/DocumentationComments/AbstractDocumentationCommentCommandHandler.cs

```diff
@@ -97,13 +125,292 @@ private bool CompleteComment(
+
+            if (paramStartTag == -1 || paramEndTag == -1)
+            {
+                break;
+            }
+
+            var paramNameStart = paramStartTag + "<param name=\"".Length;
+            var paramNameEnd = comments.IndexOf("\">", paramNameStart, StringComparison.Ordinal);
+            if (paramNameEnd != -1)
+            {
+                var parameterName = comments.Substring(paramNameStart, paramNameEnd - paramNameStart);
+                proposedEdits.Add(new DocumentationCommentProposedEdit(new TextSpan(paramEndTag + startIndex, 0), parameterName, DocumentationCommentTagType.Param));
+            }
+
+            index = paramEndTag + "</param>".Length;
+        }
+
+        var returnsEndTag = comments.IndexOf("</returns>", index, StringComparison.Ordinal);
+        if (returnsEndTag != -1)
+        {
+            proposedEdits.Add(new DocumentationCommentProposedEdit(new TextSpan(returnsEndTag + startIndex, 0), null, DocumentationCommentTagType.Returns));
+        }
+
+        while (true)
+        {
+            var exceptionEndTag = comments.IndexOf("</exception>", index, StringComparison.Ordinal);
+            var exceptionStartTag = comments.IndexOf("<exception cref=\"", index, StringComparison.Ordinal);
+
+            if (exceptionEndTag == -1 || exceptionStartTag == -1)
+            {
+                break;
+            }
+
+            var exceptionNameStart = exceptionStartTag + "<exception cref=\"".Length;
+            var exceptionNameEnd = comments.IndexOf("\">", exceptionNameStart, StringComparison.Ordinal);
+            if (exceptionNameEnd != -1)
+            {
+                var exceptionName = comments.Substring(exceptionNameStart, exceptionNameEnd - exceptionNameStart);
+                proposedEdits.Add(new DocumentationCommentProposedEdit(new TextSpan(exceptionEndTag + startIndex, 0), exceptionName, DocumentationCommentTagType.Exception));
+            }
+
+            index = exceptionEndTag + "</exception>".Length;
+        }
+
+        return new DocumentationCommentProposal(memberNode.ToFullString(), proposedEdits.ToImmutableArray());
+    }
+
+    /// <summary>
+    /// Calls into the copilot service to get the pieces for the documentation comment.
+    /// </summary>
+    private static async Task<IReadOnlyList<ProposedEdit>> GetProposedEditsAsync(
+        DocumentationCommentProposal proposal, ICopilotCodeAnalysisService copilotService,
+        ITextSnapshot oldSnapshot, string? indentText, CancellationToken cancellationToken)
+    {
+        var list = new List<ProposedEdit>();
+        var (copilotText, isQuotaExceeded) = await copilotService.GetDocumentationCommentAsync(proposal, cancellationToken).ConfigureAwait(false);
+
+        // Quietly fail if the quota has been exceeded.
+        if (isQuotaExceeded)
+        {
+            return list;
+        }
+
+        // The response from Copilot is structured like a JSON object, so make sure it is being returned appropriately.
```

Comment:
Since the format of the response is determined by the instruction in the prompt, I think it'd make more sense to keep the response parsing at the same place of prompt construction. i.e. `GetDocumentationCommentAsync` should return structured data instead of a string